Week-4/Assigment4.1/promises.js

```diff
@@ -0,0 +1,72 @@
+const STATE = {
+    PENDING: 'PENDING',
+    FULFILLED: 'FULFILLED',
+    REJECTED: 'REJECTED',
+}
+class MyPromise {
+    constructor(callback) {
+        // Initial state of Promise is empty
+        this.state = STATE.PENDING;
+        this.value = undefined;
+        this.handlers = [];
+        // Invoke callback by passing the resolve and the reject function of our class
+        callback(this.resolve, this.reject);
+    }
+
+    resolve = (value) => {
+        if (this.state !== 'PENDING') return;
+
+        this.state = 'FULFILLED'
+        this.value = value
+        console.log(this.value)
+        this.finally()
+    }
+
+    reject = (error) => {
+        if (this.state !== 'PENDING') return;
+        this.state = 'REJECTED'
+        this.value = error
+        console.log(this.value)
+        this.finally()
+    }
+
+    finally(callback) {
+        this.state = 'FULFILLED'
+        console.log("finally")
+    }
+
+
+}
+
+
+const getNumber = () => {
+
+    return Math.random() * 100
+
+
+}
+
+const testPromiseRejction = new MyPromise((res, rej) => {
+    let randomNumber = getNumber()
+    let Divisor = 5
+    setTimeout(() => {
```

Comment:
A promise without then is not a promise. Understand how promise works

Understand from this article: [Link](https://medium.com/nerd-for-tech/implement-your-own-promises-in-javascript-68ddaa6a5409)